### PR TITLE
Add snapshot interval operations to CLI

### PR DESCRIPTION
**Pull request by ip1981**
```
[root@adm ~]# iml snapshot interval list
+----+------------+---------------------+-------------+----------+
| Id | Filesystem | Interval            | Use Barrier | Last Run |
+====+============+=====================+=============+==========+
| 1  | zfsmo      | 1 minute            | false       | --       |
+----+------------+---------------------+-------------+----------+
| 3  | zfsmo      | 1 week              | false       | --       |
+----+------------+---------------------+-------------+----------+
| 4  | zfsmo      | 1 year and 11 hours | false       | --       |
+----+------------+---------------------+-------------+----------+

[root@adm ~]# iml snapshot interval add zfsmo  10 year -b
[root@adm ~]# iml snapshot interval list
+----+------------+-------------------------------+-------------+----------+
| Id | Filesystem | Interval                      | Use Barrier | Last Run |
+====+============+===============================+=============+==========+
| 1  | zfsmo      | 1 minute                      | false       | --       |
+----+------------+-------------------------------+-------------+----------+
| 3  | zfsmo      | 1 week                        | false       | --       |
+----+------------+-------------------------------+-------------+----------+
| 4  | zfsmo      | 1 year and 11 hours           | false       | --       |
+----+------------+-------------------------------+-------------+----------+
| 10 | zfsmo      | 10 years, 2 days and 12 hours | true        | --       |
+----+------------+-------------------------------+-------------+----------+

[root@adm ~]# iml snapshot interval remove 1 3
[root@adm ~]# iml snapshot interval list
+----+------------+-------------------------------+-------------+----------+
| Id | Filesystem | Interval                      | Use Barrier | Last Run |
+====+============+===============================+=============+==========+
| 4  | zfsmo      | 1 year and 11 hours           | false       | --       |
+----+------------+-------------------------------+-------------+----------+
| 10 | zfsmo      | 10 years, 2 days and 12 hours | true        | --       |
+----+------------+-------------------------------+-------------+----------+
```

Signed-off-by: Igor Pashev <pashev.igor@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2257)

### CODE REVIEW

File: iml-api/src/graphql.rs [hunk 1]
```diff
@@ -659,6 +659,8 @@ impl MutationRoot {
                     interval
                 )
                 VALUES ($1, $2, $3)
+                ON CONFLICT (filesystem_name, interval)
+                DO UPDATE SET use_barrier = EXCLUDED.use_barrier
```
Comment (jgrund) [inline iml-api/src/graphql.rs:663]: We should ```DO NOTHING``` here. If we change a param, the backing timer will need to be regenerated.

**Review by jgrund — COMMENTED** [review]
Just a comment on the conflict change, otherwise LGTM.

**Review by mkpankov — APPROVED** [review]
> Reviewed 5 of 7 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2257)* status: all files reviewed, 2 unresolved discussions (waiting on @ip1981)

___
*[iml-manager-cli/src/snapshot.rs, line 28 at r2](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2257#-MHok2vd0bPp3pJ0ioxM:-MHok2vd0bPp3pJ0ioxN:b-8vckjo) ([raw file](https://github.com/whamcloud/integrated-manager-for-lustre/blob/ac0811caac3d38ab8687bb7e29e2590cb964fff1/iml-manager-cli/src/snapshot.rs#L28)):*
> ```Rust
>         #[structopt(short = "d", long = "display", default_value = "tabular")]
>         display_type: DisplayType,
>         // TODO? Optional FS?
> ```

Still TODO?

File: iml-manager-cli/src/display_utils.rs [hunk 2]
```diff
@@ -151,6 +153,28 @@ impl IntoTable for Vec<Snapshot> {
...
+        generate_table(
+            &["Id", "Filesystem", "Interval", "Use Barrier", "Last Run"],
+            self.into_iter().map(|i| {
+                vec![
+                    i.id.to_string(),
+                    i.filesystem_name,
+                    chrono::Duration::from_std(i.interval.0)
+                        .map(HumanTime::from)
+                        .map(|x| x.to_text_en(Accuracy::Precise, Tense::Present))
+                        .unwrap_or("---".into()),
```
Comment (jgrund) [inline iml-manager-cli/src/display_utils.rs:167]: nit: clippy warning here.

**Review by jgrund — APPROVED** [review]

**Pull request merged**